### ISSUE TITLE
Huge .Otter_Log file, submission to gradescope fails

### ISSUE THREAD

**Issue by codingPie** [ucbds-infra/otter-grader#856]
I have a very simple assignment and most students made the submission and have the work graded without any problem. However, one student could not have his zip file submitted. I found that his zip file was close to 400 MB while the zip files of other students were around 5KB. His code works fine but I found that his otter_log file was close to 400 MB. I manually graded his file hoping that the problem will go away from future assignments. Now I realized that the log file is accumulative. All the events from when students started to use Otter Grader are saved in the log file. This means that this students will never be able to submit his work to GradeScope in the future. It also means that other students might also mess up their code and ended up with huge log files. I have not set "save_environment" or the "variables" setting since I am not grading from log files. I wonder:

1. How could his log file be so large?
2. Where is the log file saved? How can it be emptied?

Thanks!

**Comment by chrispyles** [comment]
Is this from a notebook that's being reused? Otter writes to .OTTER_LOG whenever a check is run. The log file is assumed to be located in whatever the working directory is when a `Notebook` method is used (which is usually the directory containing the notebook file, unless the notebook code changes it). I'm not sure how it could get to 400 MB without it being reused across many assignments or a huge number of `Notebook` methods (`check`, `check_all`, `__init__`, `export`, etc.) being called.

If you want it excluded from the submission, you'll need to delete the file before calling `Notebook.export` (e.g. add a cell before the export cell that runs `!rm .OTTER_LOG`, or add it to top of the export cell as a postprocessing step). Otter includes it if it finds the file, and doesn't currently support filtering it out (feel free to create feature request to add a configuration to stop the log from being included, though).

**Comment by codingPie** [comment]
Thanks so much for the quick help! I am not sure what it means by reusing a notebook. I did reuse the solution notebook from copying a pervious one to save the work of adding the Raw NBConvert cells. When the I have the solution notebook created, I run otter assign to create the student version notebook. That's probably why I see accumulative events from otter.log files of students with small submissions. 

Not sure why this student could manage to generate such a huge file but I will have him add !rm .OTTER_LOG and see if that is a solution. I am creating a feature request as well.

**Comment by chrispyles** [comment]
>  I did reuse the solution notebook from copying a pervious one to save the work of adding the Raw NBConvert cells. When the I have the solution notebook created, I run otter assign to create the student version notebook.

When you distribute the notebook to students, is there a .OTTER_LOG file in the directory containing the notebook?

> I am not sure what it means by reusing a notebook.

I mean having students use the same notebook file for multiple assignments, i.e. several assignments that build on each other in a single notebook file.

another thing i thought of: if there are lots of tests and the student uses "run all cells" a lot, this could add up.

**Comment by codingPie** [comment]
When you distribute the notebook to students, is there a .OTTER_LOG file in the directory containing the notebook?
-- No, I followed the tutorial of Otter Grade so my directory structure is exactly the same as the tutorial. See the attached screenshot. 
<img width="499" alt="Screenshot 2024-10-07 at 8 56 19 PM" src="https://github.com/user-attachments/assets/0f819214-2eab-4d34-90fc-64d628082f67">

I have 12 test cases for this simple assignment. I think the student must have run all cells many times. I realized last week that students don't know how to test code and were completely relying on public test cases to debug blindly. 

I did ask the student to download a new copy of the notebook and copy paste all the solutions over. I don't know whether he followed my instructions and whether he did so but used the same folder. So having a setting to remove the log file is quite important.
